freyja_trajectory_provider/src/trajectory_provider.cpp

```diff
@@ -57,6 +57,28 @@ TrajRef getCircleReference( const ros::Duration &cur_time )
   float xvel, xpos;
   ref_state.header.stamp = ros::Time::now();
   
+  ref_state.pn = 0.5*std::sin( 0.5*t );
+  ref_state.pe = 0.5*std::cos( 0.5*t );
+  ref_state.pd = -1.0;
+  
+  ref_state.vn = 0.25*std::cos( 0.5*t );
+  ref_state.ve = -0.25*std::sin( 0.5*t );
+  ref_state.vd = 0.0;
+  
+  ref_state.yaw = 0.0;
+
+  // set an, ae, ad to second derivatives if needed for FF..
+  return ref_state;
+}
+
+// CIRCLE: pn = A*sin(wt), pe = A*cos(wt), vn = A*w*cos(wt) ..
+TrajRef getAggressiveCircleReference( const ros::Duration &cur_time )
```

Comment:
Make common function: `getCircleReference( const ros::Duration &t, const int agg_level)`

Then, based on `agg_level` (1,2,3..) define values for variables `A` and `w`. `A` is amplitude (radius), `w` angular rate such that `2pi/w` = (seconds for one rev). So, `w=1` means 6.24s, `w=2` means 3.14s etc .. larger `w` ==> faster circles.

Then,
```
pn = A*std::sin( w*t );
pe = A*std::cos( w*t );
: 
vn = A*w*std::cos( w*t );
ve = -A*w*std::sin( w*t );
```